aws/src/main/java/org/apache/iceberg/aws/s3/S3URI.java

```diff
@@ -37,6 +37,9 @@ class S3URI {
   private static final String QUERY_DELIM = "\\?";
   private static final String FRAGMENT_DELIM = "#";
 
+  /** Suffix of S3Express storage bucket names. */
+  private static final String S3EXPRESS_STORE_SUFFIX = "--x-s3";
```

Comment:
can we call this DIRECTORY_BUCKET_NAME_SUFFIX to be consistent in naming? (also fix the comment line above)